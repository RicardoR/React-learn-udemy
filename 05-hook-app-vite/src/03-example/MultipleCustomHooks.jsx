import {useCounter, useFetch} from "../hooks";
import {LoadingQuote, Quote} from "./";


export const MultipleCustomHooks = () => {

    const {counter, increment} = useCounter(1)
    const {data, isLoading, hasError} = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
    const {author, quote} = !!data && data[0];

    return (
        <>
            <h1> breaking bad Quotes</h1>
            <hr/>

            {
                isLoading
                    ? <LoadingQuote></LoadingQuote>
                    : <Quote quote={quote} author={author}></Quote>
            }

            <button className='btn btn-primary'
                    disabled={isLoading}
                    onClick={() => increment()}>
                Next quote
            </button>
        </>
    );
};



