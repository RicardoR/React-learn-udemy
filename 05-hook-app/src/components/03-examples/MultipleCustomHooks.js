import React from 'react';
import { useFetch } from '../../hooks/useFetch';
import { useCounter } from '../../hooks/useCounter';
import './multiple-custom.css';

export const MultipleCustomHooks = () => {
  const { counter, increment } = useCounter(1);

  const { data, loading } = useFetch(
    `https://www.breakingbadapi.com/api/quotes/${counter}`
  );

  const { author, quote } = !!data && data[0];

  return (
    <>
      <h1>Breaking bad quotes</h1>
      <hr />
      {loading ? (
        <div className="alert alert-info text-center">Loading...</div>
      ) : (
        <blockquote className="blockquote text-right">
          <p className="mb-2">{quote}</p>
          <footer className="blockquote-footer"> {author}</footer>
        </blockquote>
      )}

      <button className="btn btn-primary" onClick={() => increment(1)}>
        Siguiente frase
      </button>
    </>
  );
};
