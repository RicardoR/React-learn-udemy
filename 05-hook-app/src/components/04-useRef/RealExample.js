import React, { useState } from 'react'
import { MultipleCustomHooks } from '../../components/03-examples/MultipleCustomHooks';

export const RealExample = () => {
    const [show, setShow] = useState(false);

    return (
      <>
        <h1>RealExample</h1>
            <hr />
            
            {show && <MultipleCustomHooks />}
            
            <p className='mt-3'>
                <button
                    className='btn btn-primary'
                    onClick={() => setShow(!show)}
                >
                Show / Hide
                </button>
            </p>
      </>
    );
}
