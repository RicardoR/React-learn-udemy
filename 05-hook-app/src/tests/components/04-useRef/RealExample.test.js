import React from 'react';
import { shallow } from 'enzyme';
import { RealExample } from '../../../components/04-useRef/RealExample';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';

describe('RealExample UT', () => {
  const wrapper = shallow(<RealExample />);

  test('should display correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(MultipleCustomHooks).exists()).toBe(false);
  });

  test('should display the <MultipleCustomHooks /> component', () => {
    const button = wrapper.find('button');
    button.simulate('click');
    expect(wrapper.find(MultipleCustomHooks).exists()).toBe(true);
  });
});
