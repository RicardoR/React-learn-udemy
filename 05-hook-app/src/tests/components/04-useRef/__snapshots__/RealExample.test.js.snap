// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RealExample UT should display correctly 1`] = `
<Fragment>
  <h1>
    RealExample
  </h1>
  <hr />
  <p
    className="mt-3"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Show / Hide
    </button>
  </p>
</Fragment>
`;
